engine.js

```diff
@@ -17,15 +19,29 @@ var filter = function(array) {
 module.exports = function (options) {
 
   var types = options.types;
-
   var length = longest(Object.keys(types)).length + 1;
-  var choices = map(types, function (type, key) {
+  var typeChoices = map(types, function (type, key) {
     return {
       name: rightPad(key + ':', length) + ' ' + type.description,
       value: key
     };
   });
 
+  var scopePrompt = {
+    type: 'input',
+    name: 'scope',
+    message: 'What is the scope of this change (e.g. component or file name)? (press enter to skip)\n'
+  };
+  var choices = require(path.join(path.resolve(''), 'package.json')).config.commitizen.scopeList;
```

Comment:
What if:
```javascript
var get = require('lodash.get')
/// ....
get(require(path.join(path.resolve(''), 'package.json')), 'config.commitizen.scopeList')
```
To avoid error if the package.json doesn't have all the object path.

Reply:
The reason I didn't put a failsafe in place is because this package will only kick in if commitizen is configured, which requires `config.commitizen` to be defined.
Then, if a first-child is referenced and it doesn't exist (like `.scopeList`), `undefined` will be returned instead of an error.
Do you think it's worth adding something like lodash to failsafe this?

Reply:
If you're sure about that, in this case, this seems good.